hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaTargetManagement.java

```diff
@@ -648,67 +620,65 @@ public long countByInstalledDistributionSet(final long distId) {
     }
 
     @Override
-    public Page<Target> findByTargetFilterQueryAndNonDS(final Pageable pageRequest, final long distributionSetId,
-            final String targetFilterQuery) {
-        throwEntityNotFoundIfDsDoesNotExist(distributionSetId);
-
-        final Specification<JpaTarget> spec = RSQLUtility.parse(targetFilterQuery, TargetFields.class,
-                virtualPropertyReplacer, database);
-
-        return findTargetsBySpec(
-                (root, cq,
-                        cb) -> cb.and(spec.toPredicate(root, cq, cb), TargetSpecifications
-                                .hasNotDistributionSetInActions(distributionSetId).toPredicate(root, cq, cb)),
-                pageRequest);
-
+    public Page<Target> findByQueryAndNotInRolloutGroups(final Pageable pageRequest, final Collection<Long> groupIds,
+            final String query) {
+        // TODO: Refactor assignment of targets to rollout-groups in order to
```

Comment:
fix this TODO

Reply:
I would prefere to do this in a sperate PR